src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+        }
+
+        // find discrete intervals that contain the breakpoint evidence
+        final Iterator<Interval> intervalItr =
+                evidenceRDD
+                        .mapPartitions(evidenceItr ->
+                                new MapPartitioner<>(evidenceItr,
+                                        new IntervalMapper(maxFragmentSize), new BreakpointEvidence(nContigs)), true)
+                        .collect()
+                        .iterator();
+
+        // coalesce overlapping intervals (can happen at partition boundaries)
+        final List<Interval> intervals = new ArrayList<>();
+        if ( intervalItr.hasNext() ) {
+            Interval prev = intervalItr.next();
+            while ( intervalItr.hasNext() ) {
+                final Interval next = intervalItr.next();
+                if ( prev.isDisjointFrom(next) ) {
+                    intervals.add(prev);
+                    prev = next;
+                } else {
+                    prev = prev.join(next);
+                }
+            }
+            intervals.add(prev);
+        }
+
+        if ( locations.evidenceDir != null ) evidenceRDD.unpersist();
+
+        // record the intervals
+        if ( locations.intervalFile != null ) {
+            try (final OutputStreamWriter writer = new OutputStreamWriter(new BufferedOutputStream(
+                    BucketUtils.createFile(locations.intervalFile, pipelineOptions)))) {
+                for (final Interval interval : intervals) {
+                    final String seqName = contigs.get(interval.getContig()).getSequenceName();
+                    writer.write(seqName + " " + interval.getStart() + " " + interval.getEnd() + "\n");
+                }
+            } catch (final IOException ioe) {
+                throw new GATKException("Can't write intervals file " + locations.intervalFile, ioe);
+            }
+        }
+
+        return intervals;
+    }
+
+    /** gather some interesting factoids about the reads in aggregate */
+    @VisibleForTesting static ReadMetadata getMetadata( final SAMFileHeader header,
+                                             final int meanBasesPerTemplate ) {
+        final List<SAMReadGroupRecord> groups = header.getReadGroups();
+        final int nGroups = groups.size();
+        final ReadMetadata.ReadGroupFragmentStatistics groupStats =
+                //TODO: get real data
+                new ReadMetadata.ReadGroupFragmentStatistics(400.f, 75.f);
+        final List<ReadMetadata.ReadGroupFragmentStatistics> stats = new ArrayList<>(nGroups);
+        for ( int idx = 0; idx != nGroups; ++idx ) {
+            stats.add(groupStats);
+        }
+        return new ReadMetadata(header, stats, groupStats, meanBasesPerTemplate);
+    }
+
+    /** mean number of sequenced bases per template (i.e., read length if unpaired, twice the read length if paired) */
+    @VisibleForTesting static int getMeanBasesPerTemplate(final JavaRDD<GATKRead> reads ) {
```

Comment:
We'll use this pass to gather the template size statistics, too, in the not-too-distant future, so it may seem less heavy at that point.  I'm guessing that the JavaRDDLike::sample method won't be faster, but I haven't tested that.  JavaRDDLike::take seems like it could be the right thing, but the documentation seems to imply that something complicated is going on.  I could write a partition mapper that takes that first N elements from each partition.  Does that seem like a good sample?